x/mongo/driver/topology/topology_options.go

```diff
@@ -7,451 +7,345 @@
 package topology
 
 import (
-	"bytes"
 	"crypto/tls"
-	"crypto/x509"
-	"encoding/pem"
 	"errors"
-	"fmt"
-	"io/ioutil"
 	"strings"
 	"time"
 
 	"go.mongodb.org/mongo-driver/event"
+	"go.mongodb.org/mongo-driver/mongo/description"
+	"go.mongodb.org/mongo-driver/mongo/options"
 	"go.mongodb.org/mongo-driver/x/mongo/driver"
 	"go.mongodb.org/mongo-driver/x/mongo/driver/auth"
 	"go.mongodb.org/mongo-driver/x/mongo/driver/connstring"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/ocsp"
 	"go.mongodb.org/mongo-driver/x/mongo/driver/operation"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/session"
 )
 
-// Option is a configuration option for a topology.
-type Option func(*config) error
-
-type config struct {
-	mode                   MonitorMode
-	replicaSetName         string
-	seedList               []string
-	serverOpts             []ServerOption
-	cs                     connstring.ConnString // This must not be used for any logic in topology.Topology.
-	uri                    string
-	serverSelectionTimeout time.Duration
-	serverMonitor          *event.ServerMonitor
-	srvMaxHosts            int
-	srvServiceName         string
-	loadBalanced           bool
+// Config is used to construct a topology.
+type Config struct {
+	Mode                   MonitorMode
+	ReplicaSetName         string
+	SeedList               []string
+	ServerOpts             []ServerOption
+	ConnString             connstring.ConnString // This must not be used for any logic in topology.Topology.
+	URI                    string
+	ServerSelectionTimeout time.Duration
+	ServerMonitor          *event.ServerMonitor
+	SRVMaxHosts            int
+	SRVServiceName         string
+	LoadBalanced           bool
 }
 
-func newConfig(opts ...Option) (*config, error) {
-	cfg := &config{
-		seedList:               []string{"localhost:27017"},
-		serverSelectionTimeout: 30 * time.Second,
-	}
+func hasOptionalData(co options.ClientOptions) bool {
+	return co.Direct != nil && *co.Direct ||
+		co.ServerMonitor != nil ||
+		co.ReplicaSet != nil ||
+		co.ServerSelectionTimeout != nil ||
+		co.LoadBalanced != nil
+}
 
-	for _, opt := range opts {
-		if opt == nil {
-			continue
-		}
-		err := opt(cfg)
-		if err != nil {
-			return nil, err
-		}
+// ConvertToDriverAPIOptions converts a options.ServerAPIOptions instance to a driver.ServerAPIOptions.
+func ConvertToDriverAPIOptions(s *options.ServerAPIOptions) *driver.ServerAPIOptions {
```

Comment:
Does function this need to be exported? If not, change it to be un-exported.

Reply:
Looks like it's called from other packages, so I think it probably needs to be exported.